src/openforms/submissions/api/viewsets.py

```diff
@@ -108,60 +102,64 @@ def _complete(self, request, *args, **kwargs):
 
         Submission completion requires that all required steps are completed.
 
-        Once a submission is completed, it's removed from the session. This means it's
-        no longer possible to change or read the submission data (including individual
-        steps).
+        Note that the processing of the submission runs in the background, and you
+        should periodically check the submission status endpoint to check the state.
+        Background processing makes sure that:
+
+        * potential appointments are registered
+        * a report PDF is generated
+        * the submission is persisted to the configured backend
+        * payment is initiated if relevant
+
+        Once a submission is done processing, it's scheduled to be removed from the
+        session. This means it's no longer possible to change or read the submission
+        data (including individual steps).
 
-        The submission is persisted to the configured backend.
+        TODO: give some leeway? add endpoint to trigger cleanup immediately?
+        TODO: emit a status URL with time-based token (Salted HMAC, from token generator)
+        that the frontend can poll without needing the submission in the session.
+        TODO: if appointment registration fails, the submission ID must be added back
+        to the session so the user can return to the first step(s) in the UI.
         """
         submission = self.get_object()
         validate_submission_completion(submission, request=request)
-        submission.completed_on = timezone.now()
 
+        submission.completed_on = timezone.now()
         submission.save()
+
+        # TODO: implement in celery tasks in cleanup (see ./tasks/__init__.py)
```

Comment:
Current train of thought is to _not_ do this in celery (it requires us to pass around the session object which is... weird), but instead keep this and re-add the submission ID to the session if the user needs to correct mistakes.